demos/text_detection_demo/cpp/src/cnn.cpp

```diff
@@ -119,3 +102,185 @@ InferenceEngine::BlobMap Cnn::Infer(const cv::Mat &frame) {
+
+    // ---------------------------------------------------------------------------------------------------
+    InputInfo::Ptr input_info = network_encoder.getInputsInfo().begin()->second;
+    input_name_ = network_encoder.getInputsInfo().begin()->first;
+
+    input_info->setLayout(Layout::NCHW);
+    input_info->setPrecision(Precision::U8);
+
+    channels_ = input_info->getTensorDesc().getDims()[1];
+    input_size_ = cv::Size(input_info->getTensorDesc().getDims()[3], input_info->getTensorDesc().getDims()[2]);
+
+    // --------------------------- Loading model to the device -------------------------------------------
+    ExecutableNetwork executable_network_encoder = ie.LoadNetwork(network_encoder, deviceName);
+    ExecutableNetwork executable_network_decoder = ie.LoadNetwork(network_decoder, deviceName);
+    // ---------------------------------------------------------------------------------------------------
+
+    // --------------------------- Creating infer request ------------------------------------------------
+    infer_request_encoder_ = executable_network_encoder.CreateInferRequest();
+    infer_request_decoder_ = executable_network_decoder.CreateInferRequest();
+    // ---------------------------------------------------------------------------------------------------
+
+    is_initialized_ = true;
+}
+
+void EncoderDecoderCNN::setInOutNames(const std::string out_enc_hidden_name,
+                                    const std::string out_dec_hidden_name,
+                                    const std::string in_dec_hidden_name,
+                                    const std::string features_name,
+                                    const std::string in_dec_symbol_name,
+                                    const std::string out_dec_symbol_name,
+                                    const std::string logits_name) {
+    out_enc_hidden_name_ = out_enc_hidden_name;
+    out_dec_hidden_name_ = out_dec_hidden_name;
+    in_dec_hidden_name_ = in_dec_hidden_name;
+    features_name_ = features_name;
+    in_dec_symbol_name_ = in_dec_symbol_name;
+    out_dec_symbol_name_ = out_dec_symbol_name;
+    logits_name_ = logits_name;
+}
+
+InferenceEngine::BlobMap EncoderDecoderCNN::Infer(const cv::Mat &frame) {
+    std::chrono::steady_clock::time_point begin = std::chrono::steady_clock::now();
+
+    cv::Mat image;
+    if (channels_ == 1) {
+         cv::cvtColor(frame, image, cv::COLOR_BGR2GRAY);
+    } else {
+        image = frame.clone();
+    }
+
+    InferenceEngine::Blob::Ptr blob = infer_request_encoder_.GetBlob(input_name_);
+    matU8ToBlob<uint8_t>(image, blob);
+    // --------------------------- Doing inference -------------------------------------------------------
+    /* Running the request synchronously */
+    infer_request_encoder_.Infer();
+    // ---------------------------------------------------------------------------------------------------
```

Comment:
Could you merge these comments, or remove them: `Infer()` is descriptive by itself but comments explain the same. Same for `Cnn`.

Reply:
Done